### PR TITLE
feat: add is external logic from NuxtLink component

### LINKED ISSUE #2272: NuxtLinkLocale behaves differently than NuxtLink for external links
### Environment

- Operating System: Linux
- Node Version:     v16.20.0
- Nuxt Version:     3.6.5
- Nitro Version:    2.5.2
- Package Manager:  pnpm@8.6.7
- Builder:          vite
- User Config:      extends, i18n
- Runtime Modules:  -
- Build Modules:    -

### Reproduction

```vue
<NuxtLink to="http://localhost:8080/">
  NuxtLink http://localhost:8080/
</NuxtLink>
<NuxtLinkLocale to="http://localhost:8080/">
  NuxtLinkLocale http://localhost:8080/
</NuxtLinkLocale>
```

### Describe the bug

I would also expect NuxtLinkLocale to behave like NuxtLink for an external link and not simply redirect to '/'. Otherwise you would have to build an extra component that checks if it is an external link and then uses NuxtLink or NuxtLinkLocale (and the whole point of NuxtLinkLocale was that you don't have to build your own component).

Output:
```html
<a href="http://localhost:8080/" rel="noopener noreferrer">
  NuxtLink http://localhost:8080/
</a>
<a aria-current="page" href="/" class="router-link-active router-link-exact-active">
  NuxtLinkLocale http://localhost:8080/
</a>
```

### Additional context

_No response_

### Logs

_No response_

### PR DESCRIPTION
**Pull request by TitusKirch**
### 🔗 Linked issue
#2272

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Resolves #2272

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by kazupon — APPROVED** [review]
Thank you for your contribution!
Look good to me.

@BobbieGoede 
Could you review this PR, please? 🙏

File: src/runtime/components/NuxtLinkLocale.ts [hunk 1]
```diff
@@ -23,8 +25,32 @@ export default defineComponent({
...
+      if (typeof props.to === 'object') {
+        return false;
+      }
+
+      return props.to === '' || hasProtocol(props.to, { acceptRelative: true });
+    });
+
+    return () => isExternal.value
+        ? h(NuxtLink, props, slots.default)
+        : h(NuxtLinkLocale, { ...props, to: resolvedPath }, slots.default);
```
Comment (BobbieGoede) [inline src/runtime/components/NuxtLinkLocale.ts]: `NuxtLinkLocale` could be used for both returns, but I'm not really sure if it makes any practical difference 😅.

File: src/runtime/components/NuxtLinkLocale.ts [hunk 2]
```diff
@@ -1,5 +1,7 @@
+import { NuxtLink } from '#components';
```
Comment (BobbieGoede) [inline src/runtime/components/NuxtLinkLocale.ts]: It seems like the linter doesn't check these files, note the semicolons and line length changes.

**Review by BobbieGoede — APPROVED** [review]
Thank you for your contribution! My review feels a bit nitpicky, your PR gets my approval either way 😄

File: src/runtime/components/NuxtLinkLocale.ts (hunk 1 above)
Comment (TitusKirch) [reply src/runtime/components/NuxtLinkLocale.ts]: Thank you for your review! I don't think this makes a practical difference, but I understand that using `NuxtLinkLocale` for both return values would be more likely to meet the expectation.

**Comment by BobbieGoede** [comment]
Thanks again for contributing! 🚀❤️

**Pull request merged**